chainforge/react-server/src/backend/ai.ts

```diff
@@ -396,6 +409,82 @@ export async function autofillTable(
   }
 }
 
+/**
+ * Uses an LLM to generate one new column with data based on the pattern explained in `prompt`.
+ * @param prompt Description or pattern for the column content.
+ * @param provider The LLM provider to use (e.g., OpenAI, Bedrock).
+ * @param apiKeys API keys required for the LLM query.
+ * @returns A promise resolving to an array of strings (column values).
+ */
+export async function generateColumn(
+  tableData: { cols: string[]; rows: Row[] },
+  prompt: string,
+  provider: string,
+  apiKeys: Dict,
+): Promise<{col: string; rows: string[]}> {
+
+  // Build a unique ID based on the input prompt
+  const id = JSON.stringify([prompt]);
+
+  // Encode the input table to a markdown table
+  const encoded = encodeTable(tableData.cols, tableData.rows);
+
+  // Extract template variables from the columns and rows
+  const templateVariables = [
+    ...new Set([
+      ...new StringTemplate(tableData.rows.join("\n")).get_vars(),
+      ...new StringTemplate(tableData.cols.join("\n")).get_vars(),
+    ]),
+  ];
+
+  // History for the system message to set up LLM behavior
+  const history: ChatHistoryInfo[] = [
+    {
+      messages: [
+        {
+          role: "system",
+          content: generateColumnSystemMessage(templateVariables, prompt),
+        },
+      ],
+      fill_history: {},
+    },
+  ];
+
+  try {
+    // Query the LLM
+    const result = await queryLLM(
+      id,
+      getAIFeaturesModels(provider).small, // Use the small model
+      1,
+      encoded,
+      {},
+      history,
+      apiKeys,
+      true,
+    );
+
+    // Handle any errors in the response
+    if (result.errors && Object.keys(result.errors).length > 0) {
+      throw new Error(Object.values(result.errors)[0].toString());
+    }
+
+    console.log("LLM said: ", result.responses[0].responses[0]);
+
+    // Decode the LLM response as a column in a markdown table
+    const columnValues = decodeTable(result.responses[0].responses[0] as string);
```

Comment:
So this is dangerous, because it relies upon the LLM to exactly reproduce the table. As I mentioned in our Discord conversation, you should instead batch the table into say, 5-row batches, then ask the LLM for a list of items that represents the items in the new column, for each item in the table. Importantly, in this solution, you do not ask the LLM to reproduce the table. This solution is also scalable because you have to imagine the tables in ChainForge can be quite large ---say, 1000 rows. Inputting 1000 rows directly into the context will likely not work; even if it does it will not be able to reproduce the table exactly, and even then, even if it can, it's a waste of $$. 

So, batch into a series of 5 or 10-row batches, ask for the items as a list, then loop over this until complete.